src/components/Content.tsx

```diff
@@ -0,0 +1,264 @@
+import { useState, useEffect } from "react";
+import { IPlaylist, ITrack } from "../interfaces";
+import Api from '../utils/api';
+import PlaylistWithButton from "./PlaylistWithButton";
+import PlaylistWithoutButton from "./PlaylistWithoutButton";
+import PlaylistsSection from "./PlaylistsSection";
+import Track from "./Track";
+
+/**
+ * Компонент для отображения main
+ * @component
+ * @param {string} props.searchValue - Строка, на основе которой происходит поиск плейлистов 
+ * (получена из поля input, находящемся в header)
+*/
+export default function Content(props: {
+    searchValue: string
+}) {
+    const [savedPlaylists, setSavedPlaylists] = useState<IPlaylist[]>([]);
+    const [foundPlaylists, setFoundPlaylists] = useState<IPlaylist[]>([]);
+    const [token, setToken] = useState('');
+    const [inputValue, setInputValue] = useState<string>('');
+    const [foundTracks, setFoundTracks] = useState<ITrack[]>([]);
+  
+    useEffect(() => {
+        getToken();
+    }, [])
+
+    const getToken = async () => {
+        const token = await Api.getToken();
+        setToken(token);
+    }
+
+    useEffect(() => {
+        let isCancelled = false;
+        if (token){
+            Api.getMyPlaylists(token).then((playlists: void|IPlaylist[]) => {
+                if (isCancelled) {
+                    return;
+                }
+                else if (playlists){
+                    setSavedPlaylists(playlists);
+                }
+            });
+        }
+        return () => {
+            isCancelled = true;
+        };
+    }, [token]);
+
+    useEffect(() => {
+        if (props.searchValue.length > 1) {
```

Comment:
Можно сделать проще
```suggestion
        if (props.searchValue.length) {
```

Также сейчас есть проблема. Сейчас на ввод каждого символа шлется запрос.
Тут лучше отправлять запрос с задержкой, например 150мс. И если мы успели что-то ввести, то отменяем отправку запроса.
Это можно сделать через setTimeout